includes/admin/helper/class-wc-helper.php

```diff
@@ -1149,9 +1149,19 @@ public static function get_local_woo_plugins() {
 			require_once ABSPATH . 'wp-admin/includes/plugin.php';
 		}
 
-		// Reset plugin cache before retrieving plugin list.
-		wp_clean_plugins_cache();
-		$plugins     = get_plugins();
+		$plugins = get_plugins();
+
+		/**
+		 * Check if plugins have WC headers, if not then clear cache and fetch again.
+		 * WC Headers will not be present if `wc_enable_wc_plugin_headers` hook was added after a `get_plugins` call -- for example when WC is activated/updated.
+		 * Also, get_plugins call is expensive so we should clear this cache very conservatively.
+		 */
+		$first_plugin = array_keys( $plugins )[0];
```

Comment:
It could be worth adding a check if `$plugins` is not empty.

Reply:
I am hoping that `$plugins` will at least contain WooCommerce. Is there an edge case where this won't happen?